### PR TITLE
For #164: protected_method pattern + tests have been added

### PR DESCRIPTION
**Pull request by dz-s**
#164

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#245](https://github.com/yegor256/aibolit/issues/245) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/0175a1b5-1b2d-41c2-9c45-bc49fecc873c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: test/patterns/protected_method/test_protected_method.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from pathlib import Path
+from aibolit.patterns.protected_method.protected_method import ProtectedMethod
+
+
+class TestProtectedMethod(TestCase):
+    dir_path = Path(os.path.realpath(__file__)).parent
+
+    def test_not_find_protected_method(self):
+        self.assertEqual([], ProtectedMethod().value(Path(self.dir_path, 'NoProtectedMethod.java')),
+                         'Should not match pattern protected method')
```
Comment (lyriccoder) [inline test/patterns/protected_method/test_protected_method.py:12]: add test for inner classes and for anonymous classes

**Comment by acheshkov** [comment]
@rultor merge

**Comment by dz-s** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 33 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @lyriccoder[/z](https://www.0crat.com/u/lyriccoder)[/z](https://www.0crat.com/u/lyriccoder) didn't make CR comments

**Pull request merged**